cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Replica.java

```diff
@@ -123,51 +124,58 @@ void clearLoad() {
    * (2) Clear and get the outbound network load associated with leadership from the given replica.
    * (3) Clear and get the CPU leadership load associated with leadership from the given replica.
    *
-   * @return Removed leadership load by snapshot time -- i.e. outbound network and fraction of CPU load by snapshot time.
+   * @return Removed leadership load by windows -- i.e. outbound network and fraction of CPU load by windows.
    */
-  Map<Resource, double[]> makeFollower() {
+  AggregatedMetricValues makeFollower() {
     // Remove leadership from the replica.
     setLeadership(false);
-    // Clear and get the outbound network load associated with leadership from the given replica.
-    double[] leadershipNwOutLoad = _load.loadFor(Resource.NW_OUT).doubleArray();
-    MetricValues leadershipCpuLoad = _load.loadFor(Resource.CPU);
 
+    // Get the inbound/outbound network and cpu load associated with leadership from the given replica.
+    // All the following metric values are in a shared mode to avoid data copy.
+    int cpuMetricId = KafkaMetricDef.resourceToMetricIds(Resource.CPU).get(0);
+    AggregatedMetricValues leadershipNwOutLoad = _load.loadFor(Resource.NW_OUT, false);
+    AggregatedMetricValues leadershipNwInLoad = _load.loadFor(Resource.NW_IN, false);
+    MetricValues cpuLoad = _load.loadFor(Resource.CPU, false).valuesFor(cpuMetricId);
+
+    // Create a leadership load delta to store the load change.
+    AggregatedMetricValues leadershipLoadDelta = new AggregatedMetricValues();
+    // We need to add the NW_OUT values to the delta before clearing the metric.
+    leadershipLoadDelta.add(leadershipNwOutLoad);
     // Remove the outbound network leadership load from replica.
     _load.clearLoadFor(Resource.NW_OUT);
-    double[] followerCpuLoad = new double[_load.numWindows()];
-    double[] cpuLoadChange = new double[_load.numWindows()];
 
-    for (int i = 0; i < leadershipCpuLoad.length(); i++) {
-      double newCpuLoad = ModelUtils.getFollowerCpuUtilFromLeaderLoad(_load.loadFor(Resource.NW_IN).get(i),
-                                                                      _load.loadFor(Resource.NW_OUT).get(i),
-                                                                      leadershipCpuLoad.get(i));
-      followerCpuLoad[i] = newCpuLoad;
-      cpuLoadChange[i] = leadershipCpuLoad.get(i) - newCpuLoad;
+    // compute the cpu delta.
```

Comment:
Would it be more clear if we move `compute the cpu delta` logic to a private method?